### ISSUE TITLE
Dragon cannot operate file upload button

### ISSUE THREAD

**Issue by selfthinker** [alphagov/reported-bugs#35]
**Upstream bug:** [Ref no 170922-000445](http://nuance.custhelp.com/app/account/questions/detail/i_id/2940832) (not public!)
**Date:** 2017-09-20 (?)
**Reported by:** Anika Henke
**Related to:** n/a

---

### Overview

The bug report can be accessed with the GDS internal "nuance" google group account and can only be seen when you're logged in with that user.

The content of that is:

> You cannot operate the file upload button with Dragon 13 in IE 11 or Chrome.
(It works in Firefox. And it obviously works with mouse grid.)
You can try it out in this reduced test case: http://jsbin.com/dezapi
> 
> If you think the bug is in the browser and not in Dragon, please let me know and I'm happy to report it for those browsers as well.

**Comment by selfthinker** [comment]
We only got an automated response:

> Dear Accessibility Team, 
> 
> Thank you for providing feedback about your Nuance product. 
> 
> - Your bug report will be reviewed for reproducibility 
> - Bugs reported will be triaged for fixing in a future update or version release of the product 
> - No response will be provided regarding your bug report 
> - If you require a response from Nuance, please open a new question and select a Support category 
> - Support categories include: Error Message, Hardware - Setup/Config, How to use Product, Product - Problem/Issue, Product - Setup/Config 
> 
> We appreciate your feedback and help toward improving our products. 
> 
> Best Regards 
> 
> Desktop Support & Service 
> Nuance Communication Inc.

So, it looks like they never give proper feedback to reports and we will never know how and if the status of the report progresses (unless it gets fixed and ends up in their public changelog).

**Comment by selfthinker** [comment]
I have tested this again in Dragon 15, 3 years after I first reported it, and found it is still an issue in IE as well as Chrome. (It used to work in Firefox but Dragon doesn't work with Firefox anymore.)

I have created a new bug report but Nuance doesn't think it's a bug because websites are apparently generally not supported.
They said:

> Dragon is not compatible with web pages and web based services, this is not supported!

Reference number of the new bug report: [200915-000032](http://nuance.custhelp.com/app/account/questions/detail/i_id/3765407) (the link is not public)

**Comment by hannalaakso** [comment]
Interestingly, a user reported to us that they were able to operate the file upload button with Dragon. We've asked them add the exact version of Dragon to this issue if they're able to discover it. See https://github.com/alphagov/govuk-frontend/issues/2065#issuecomment-814884493.

**Comment by querkmachine** [comment]
This issue was raised again in a recent DAC audit of Design System components and patterns, indicating that it's still not supported during typical usage of Dragon.

It is possible to interact with file inputs by using Dragon in mouse grid mode. It sounds like it's also possible by emulating keyboard-like usage with voice commands (i.e. saying "Press Tab key" and "Press Enter" to focus and activate the input).
